src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiSupportDef.java

```diff
@@ -57,6 +57,10 @@ default boolean isUnsupportedDML() {
     return !insert() || !read() || !filter();
   }
 
+  default boolean isUnsupportedInsert() {
+    return !insert();
+  }
+
```

Comment:
According to Aaron's design, it may not be a good idea to add an extra method here, otherwise, we need to create a method for each combination. [Here](https://github.com/stargate/data-api/blob/5c9863860960789ea00365ac6e60a074fe1d242d/src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiColumnDefContainer.java#L145) is the example of how Aaron use `NO_MATCHES` for different combinations.

Reply:
That example does not seem to support negation but only is-supported so that wouldn't work as-is I think. But I can probably change this to user `Predicate.negate()` or so to avoid adding new methods.
Thanks!

Reply:
@Hazel-Datastax here goes: https://github.com/stargate/data-api/pull/1805 -- simpler code, can just use simple lambda; I am not sure I understand why `Matcher` is added but it's not needed in this case (and it has no `negate()` method anyway).
Thanks!

This might be something to discuss wrt Java closures/lambdas chapter of "Java for Impatienet"; there are multiple ways to achieve the result.